packages/mobile/src/screens/tip-artist-screen/SendTipScreen.tsx

```diff
@@ -50,12 +64,112 @@ const parseToBNWei = (tipAmount: StringAudio) => {
 export const SendTipScreen = () => {
   const styles = useStyles()
   const [tipAmount, setTipAmount] = useState('')
-  const accountBalance = useSelectorWeb(getAccountBalance)
+  const accountBalance = (useSelectorWeb(getAccountBalance) ??
+    new BN('0')) as BNWei
   const navigation = useNavigation<TipArtistNavigationParamList>()
   const dispatchWeb = useDispatchWeb()
 
-  const hasInsufficientBalance =
-    accountBalance && parseToBNWei(tipAmount).gt(accountBalance)
+  const account = useSelectorWeb(getAccountUser)
+  const supportersMap = useSelectorWeb(getOptimisticSupporters)
+  const supportingMap = useSelectorWeb(getOptimisticSupporting)
+  const receiver = useSelectorWeb(getSendUser)
+  const [
+    amountToTipToBecomeTopSupporter,
+    setAmountToTipToBecomeTopSupporter
+  ] = useState<Nullable<BNWei>>(null)
+  const [supportingAmount, setSupportingAmount] = useState<Nullable<StringWei>>(
+    null
+  )
+  const [topSupporter, setTopSupporter] = useState<Nullable<Supporter>>(null)
+  const [isFirstSupporter, setIsFirstSupporter] = useState(false)
+
+  const tipAmountWei = parseToBNWei(tipAmount)
+  const hasInsufficientBalance = tipAmountWei.gt(accountBalance)
+
+  /**
+   * Get supporting info if current user is already supporting receiver
+   * so that the already supported amount can be used to determine
+   * how much is left to tip to become top supporter
+   */
+  useEffect(() => {
+    if (!account || !receiver) return
+    if (supportingAmount) return
+
+    const supportingForAccount = supportingMap[account.user_id] ?? null
+    const accountSupportingReceiver =
+      supportingForAccount?.[receiver.user_id] ?? null
+    if (accountSupportingReceiver) {
+      setSupportingAmount(accountSupportingReceiver.amount)
+    } else {
+      dispatchWeb({
```

Comment:
Maybe update web to dispatch that action too so both web and mobile do the same thing? https://github.com/AudiusProject/audius-client/pull/1415/files#diff-a18b6f1a60d7dfb83854657981e14cbab2fe2b5f9f0dbcbaee95e9d603770655R94

Reply:
hmm interesting. yeah i like it